src/gpu/save_and_compare_cpu_vs_gpu.c

```diff
@@ -39,6 +39,16 @@
 
 #define MAX(a, b) (((a) > (b)) ? (a) : (b))
 
+/************************************************************************
+
+helper functions for debugging
+
+please uncomment if needed...
+
+*************************************************************************/
+
+
+/*
```

Comment:
hi Elliott,

why do you think it’s safer?

usually for the SPECFEM codes, we do not advocate to use pre-processor directives. although it’s much more common in C-code, things tend to get harder and harder to read (especially with nested ones).  so i prefer using comments here, unless there would be some big advantage for using the directive. 

note also that standard code text editors (Xcode in my case, but also vim, emacs, etc…) will highlight comments, but they won’t highlight code within unused blocks of directives as you propose. thus looking at the highlighted code using the commented-out way is easier, than spotting which directive has a 0 or 1 and will get used or not.

cheers,
daniel



> On Jul 30, 2015, at 11:47 PM, Elliott Sales de Andrade <notifications@github.com> wrote:
> 
> In src/gpu/save_and_compare_cpu_vs_gpu.c <https://github.com/geodynamics/specfem3d_globe/pull/425#discussion_r35918459>:
> 
> > @@ -39,6 +39,16 @@
> >  
> >  #define MAX(a, b) (((a) > (b)) ? (a) : (b))
> >  
> > +/************************************************************************
> > +
> > +helper functions for debugging
> > +
> > +please uncomment if needed...
> > +
> > +*************************************************************************/
> > +
> > +
> > +/*
> It would be safer to use #if 0 here rather than block comments.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/geodynamics/specfem3d_globe/pull/425/files#r35918459>.
>